### PR TITLE
Allow env vars to be set for mule package task

### PR DESCRIPTION
**Pull request by btoll**
### Summary
This allows the BRANCH, CHANNEL and VERSION env vars to be specified in the mule yaml file.  It is the same functionality that was added later to other mule tasks.

### Test Plan
Add any or all of the above to the env block in any mule yaml file.

### CODE REVIEW

**Comment by tsachiherman** [comment]
Ben, you'll need to create a linked issue and use the template for the description before this PR could be merged in.

**Comment by btoll** [comment]
> Ben, you'll need to create a linked issue and use the template for the description before this PR could be merged in.

Done, thanks for the reminder.

File: scripts/release/mule/package/linux/amd64/deb/package.sh [hunk 1]
```diff
@@ -16,24 +16,21 @@ if [ -z "$OS_TYPE" ] || [ -z "$ARCH" ] || [ -z "$WORKDIR" ]; then
     exit 1
 fi
 
-REPO_DIR="$WORKDIR"
-BRANCH=$("$REPO_DIR/scripts/compute_branch.sh")
-CHANNEL=$("$REPO_DIR/scripts/compute_branch_channel.sh" "$BRANCH")
-BASE="$REPO_DIR/tmp/node_pkgs/$OS_TYPE/$ARCH"
+BRANCH=${BRANCH:-$(git rev-parse --abbrev-ref HEAD)}
```
Comment (egieseke) [inline scripts/release/mule/package/linux/amd64/deb/package.sh:19]: Is this changing the way that the branch is computed, in addition to accepting it as an argument?

Comment (btoll) [reply scripts/release/mule/package/linux/amd64/deb/package.sh:19]: I'll change that back to reference the script.  Initially, I had the idea to simplify as much as I could and avoid referencing other scripts, but then I forgot to change that back.  I will now.

Thanks for catching that.

Comment (btoll) [reply scripts/release/mule/package/linux/amd64/deb/package.sh:19]: And yes, it first defaults to BRANCH if it's set.

**Comment by btoll** [comment]
> This seems okay, but also includes not outputting to subdirectory /pkg? Just verifying that's intended in this PR.

Yes, that's intended.  The packages will go into the root of `/tmp/node_pkgs/linux/amd64`.

**Pull request merged**